sdk/core/core-client/src/serviceClient.ts

```diff
@@ -104,7 +104,7 @@ export class ServiceClient {
   constructor(options: ServiceClientOptions = {}) {
     this._requestContentType = options.requestContentType;
     this._baseUri = options.baseUri;
-    this._httpsClient = options.httpsClient || new DefaultHttpsClient();
+    this._httpsClient = options.httpsClient || getCachedDefaultHttpsClient();
```

Comment:
I couldn't see any perceivable difference in performance when I ran the tests at https://github.com/Azure/azure-sdk-for-js/commit/6bb44f9ce5fcab9a092a5257dd294a21735d4330 with smaller durations such as 10, 30, 60 seconds.

Reply:
Did you compare the results before and after the PR?  After the PR I expected all 3 scenarios to have similar throughput, but before the PR I expect the scenario which creates a new client for each request to be much slower and it might even crash due to client port exhaustion.

Reply:
[Discussed Offline, mentioning here for visibility]

These numbers are before Jeff's PR.
One more thing I realized is that the storage libraries rely on the caching already, which might have been the reason that I didn't see any difference.
Right now, updating the new tests to use core-http client directly to be able to test the fix.

Reply:
Update:
There is a four fold improvement in the "new client per request" case in terms of the number of operations per second on my local machine.


Before the fix | Size (bytes) | Parallel | Duration | Warmup(Ops/sec) | Iter 1(Ops/sec) | Iter 2(Ops/sec) | Iter 3(Ops/sec)
-- | -- | -- | -- | -- | -- | -- | --
CoreHTTPSingleClientTest | 10240 | 16 | 30 | 139.71 | 163.10 | 165.10 | 163.53
CoreHTTPMultiClientsInParallelTest | 10240 | 16 | 30 | 159.72 | 161.41 | 162.01 | 161.63
CoreHTTPNewClientPerReqTest | 10240 | 16 | 30 | 39.39 | 39.74 | 39.85 | 39.67


After the fix | Size (bytes) | Parallel | Duration | Warmup(Ops/sec) | Iter 1(Ops/sec) | Iter 2(Ops/sec) | Iter 3(Ops/sec)
-- | -- | -- | -- | -- | -- | -- | --
CoreHTTPSingleClientTest | 10240 | 16 | 30 | 149.76 | 159.63 | 157.00 | 159.11
CoreHTTPMultiClientsInParallelTest | 10240 | 16 | 30 | 158.98 | 152.44 | 161.63 | 161.34
CoreHTTPNewClientPerReqTest | 10240 | 16 | 30 | 157.99 | 160.39 | 160.77 | 161.24


Before - https://github.com/HarshaNalluru/azure-sdk-for-js/tree/harshan/perf/jeff-fix-test-before
After - https://github.com/HarshaNalluru/azure-sdk-for-js/tree/harshan/perf/jeff-fix-test-after